perceval/backends/git.py

```diff
@@ -632,25 +660,46 @@ def log(self, from_date=None, encoding='utf-8'):
 
         :raises RepositoryError: when an error occurs fetching the log
         """
+
         cmd_log = ['git', 'log', '--raw', '--numstat', '--pretty=fuller',
                    '--decorate=full', '--all', '--reverse', '--topo-order',
                    '--parents', '-M', '-C', '-c', '--remotes=origin']
-
         if from_date:
             dt = from_date.strftime("%Y-%m-%d %H:%M:%S %z")
             cmd_log.append('--since=' + dt)
+        env = {'LANG' : 'C', 'PAGER' : ''}
 
-        gitlog = self._exec(cmd_log, cwd=self.dirpath,
-                            env={'LANG' : 'C', 'PAGER' : ''})
-        gitlog = gitlog.split(b'\n')
-
+        self.failed_message = None
+        logging.debug("Running command %s (cwd: %s, env: %s)",
+                      ' '.join(cmd_log), self.dirpath, str(env))
+        try:
+            self.proc = subprocess.Popen(cmd_log,
+                                        stdout=subprocess.PIPE,
+                                        stderr=subprocess.PIPE,
+                                        cwd=self.dirpath,
+                                        env=env)
+        except OSError as e:
+            raise RepositoryError(cause=str(e))
+        # Spawn the thread for reading stderr
+        err_thread = threading.Thread (target = self._read_stderr, daemon = True)
```

Comment:
Don't use white spaces between the class and the `(`. Same comment for keyword arguments.